### PR TITLE
[20655] Fix flakey Log tests

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

Test `LogMacrosTests.default_macros_test` has been sporadically failing in Github CI, especially on Windows.
Analysis showed it was due to a race condition in `HELPER_WaitForEntries`, and the test not accounting for the value of `FASTDDS_ENFORCE_LOG_INFO`.

This PR fixes the race condition in `HELPER_WaitForEntries` by early returning the collection of log entries, and fixes `default_macros_test` by always waiting for 3 entries and then checking that the expected number of entries is returned, also taking into consideration the value of `FASTDDS_ENFORCE_LOG_INFO`.

@Mergifyio backport 2.13.x 2.10.x 2.6.x

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [x] Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/unittest/logging/log_macros/LogMacrosDefaultTests.cpp [hunk 1]
```diff
@@ -48,15 +48,20 @@ TEST_F(LogMacrosTests, default_macros_test)
...
 #else
-    static constexpr unsigned int expected_result = 2;
+    constexpr unsigned int expected_result = 2;
 #endif // debug macros check
 
-    auto consumedEntries = HELPER_WaitForEntries(expected_result);
+    // Warning and error should always be logged.
+    // Info might be logged depending on the log configuration.
+    // We always wait for 3 entries, though not all of them might be present.
+    auto consumedEntries = HELPER_WaitForEntries(3u);
```
Comment (Mario-DL) [inline test/unittest/logging/log_macros/LogMacrosDefaultTests.cpp:64]: Would the test be more clear using  
```suggestion
    mockConsumer->wait_for_at_least_entries(expected_result);
```
Modifying that method to return a `size_t` to assert after. What do you think ?

Comment (MiguelCompany) [reply test/unittest/logging/log_macros/LogMacrosDefaultTests.cpp:64]: There are other tests using the same method, and some of them check the content of the returned entries

**Review by Mario-DL — APPROVED** [review]
LGTM with green CI

**Comment by Mario-DL** [comment]
@richiprosima please test this

**Comment by EduPonz** [comment]
@richiprosima please test this

**Comment by EduPonz** [comment]
@Mergifyio backport 2.13.x 2.10.x 2.6.x

**Pull request merged**